packages/medusa/src/utils/api/response-compression.ts

```diff
@@ -0,0 +1,41 @@
+import { Request, Response, NextFunction } from "express"
+import compression from "compression"
+import { Logger } from "@medusajs/types"
+import {
+  ProjectConfigOptions,
+  ResponseCompressionOptions,
+} from "@medusajs/types"
+
+export function shouldCompressResponse(req: Request, res: Response) {
+  const logger: Logger = req.scope.resolve("logger")
+  const {
+    projectConfig: { response_compression_enabled },
+  } = req.scope.resolve("configModule")
+
+  if (response_compression_enabled !== true) {
```

Comment:
suggestion: Should we check for false specifically? by default we compress unless the user set it to false. Depends on my previous comment